k8s/admission.go

```diff
@@ -84,104 +54,65 @@ var now = time.Now
 // Mutate runs the underlying MutateFunc() function (if non-nil), and if that returns successfully,
 // appends additional patch operations to the MutatingResponse for CommonMetadata fields not in kubernetes standard metadata,
 // and labels internally used by the SDK, such as the stored version.
-func (o *OpinionatedMutatingAdmissionController) Mutate(request *resource.AdmissionRequest) (*resource.MutatingResponse, error) {
+func (o *OpinionatedMutatingAdmissionController) Mutate(ctx context.Context, request *resource.AdmissionRequest) (*resource.MutatingResponse, error) {
 	// Get the response from the underlying controller, if it exists
 	var err error
 	var resp *resource.MutatingResponse
-	if o.MutateFunc != nil {
-		resp, err = o.MutateFunc(request)
+	if o.Underlying != nil {
+		resp, err = o.Underlying.Mutate(ctx, request)
 		if err != nil {
 			return resp, err
 		}
 	}
-	if resp == nil || resp.PatchOperations == nil {
+	if resp == nil || resp.UpdatedObject == nil {
 		resp = &resource.MutatingResponse{
-			PatchOperations: make([]resource.PatchOperation, 0),
+			UpdatedObject: request.Object,
```

Comment:
Not sure if this is an issue but it's possible that we want to copy the object here instead of using a reference to it, so that we don't mutate the request object – I don't know if anything else depends on it.

Reply:
Fair enough 👍